File: 2015/Day 01/day01.py
#
#  Advent of Code 2015
#  Day 01
#
#  By PatchesTheDipstick
#

def getInput(file_name):
	input = []
	with open(file_name) as input_file:
		input = input_file.read.splitlines()
	return input

if __name__ == '__main__':
	
	input = getInput("input.txt")
	floor = 0
	first_basement = -1
	for index, char in enumerate(input[0]):
		if char == "(":
			floor += 1
		else:
			floor -= 1
		if (floor < 0) and (first_basement == -1):
			first_basement = index
			
	print("Advent of Code 2015 Day 01")
	print("  Part one:", floor)
	print("  Part two:", first_basement)

File: 2015/Day 01/day01.java
//
//  Advent of Code 2015
//  Day 01
//
//  By PatchesTheDipstick
//

import java.io.FileReader;
import java.lang.StringBuilder;
import java.util.Scanner;


class AoC2015Day01
{
	
	public static void main(String[] args)
	{
		String input;
		try
		{
			Scanner inFile = new Scanner(new FileReader("input.txt"));
			
			StringBuilder inBuilder = new StringBuilder();
			while (inFile.hasNext())
			{
				inBuilder.append(inFile.next());
			}
			inFile.close();
			input = inBuilder.toString();
		} catch (Exception e)
		{
			System.out.println("Oops...");
			return;
		}
		int floorEndedOn = 0;
		int pointWentUnderground = 0;
		int currFloor = 0;
		boolean hasGoneUnder = false;
		for (int index = 0; index < input.length(); ++index)
		{
			if (input.charAt(index) == '(')
				currFloor++;
			else
				currFloor--;
			if (!hasGoneUnder && currFloor < 0)
			{
				hasGoneUnder = true;
				pointWentUnderground = index + 1;
			}
		}
		floorEndedOn = currFloor;
		
		System.out.println("Advent of Code 2015 Day 01:");
		System.out.println("    Answer one: " + floorEndedOn);
		System.out.println("    Answer two: " + pointWentUnderground);
	
	}

}